geode-management/src/main/java/org/apache/geode/management/configuration/DiskStore.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.geode.management.configuration;
+
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.management.runtime.DiskStoreInfo;
+
+@Experimental
+public class DiskStore extends GroupableConfiguration<DiskStoreInfo> {
```

Comment:
If `DiskStoreInfo` is an empty class (no runtime info to gather for this configuration object), then you can declare this `DiskStore` as `public class DiskStore extends GroupableConfiguration<RuntimeInfo>`. This way, for get/list, the cms will not even issue the functions to the servers to try to retrieve an empty object.

Reply:
In its current configuration it does list the servers to which the `DiskStore` has been applied. I'll try and see if this continues when we roll it into the `DiskStore` class. If that is the case we can drop the `DiskStoreInfo` class.

Reply:
It appears the `DiskStoreRealizer` needs a concrete implementation of `RuntimeInfo`.

Reply:
If it returns null in the `get` method?

Reply:
That would work but then we wouldn't even get a list of machines where the disk-stores live at run time.

Reply:
Ah, I see, you don't need further diskstore info on each server, but you do need the server id. I guess you can leave it as is.